sound/pci/hda/patch_hdmi.c

```diff
@@ -2833,10 +2833,17 @@ static int alloc_intel_hdmi(struct hda_codec *codec)
 /* parse and post-process for Intel codecs */
 static int parse_intel_hdmi(struct hda_codec *codec)
 {
+	int retry_count = 0;
 	int err;
 
+again:
 	err = hdmi_parse_codec(codec);
 	if (err < 0) {
+		if (retry_count++ < 3) {
```

Comment:
it's better to initialize the retry line 2836 to a value and decrement it to zero, easier to figure out what the values are.